Parse/src/main/java/com/parse/ManifestInfo.java

```diff
@@ -173,14 +167,14 @@ private static boolean usesPushBroadcastReceivers() {
       intentsRegistered++;
     }
 
-    if (intentsRegistered != 0 && intentsRegistered != NUMBER_OF_PUSH_INTENTS) {
+    if (intentsRegistered != 0 && intentsRegistered != 3) {
```

Comment:
why delete this constant?

Reply:
I had my brain in cleanup mode :) and this looked already clear from the method implementation to me. I can add again.

I can also add some tests to the patch but first wanted to know if there's consensus over this being merged

Reply:
PR works for me.  It looks like you removed PPNS while also refactoring to a PushHandler interface with an eye towards allowing the FCM handler?   The only thing is that FCM/GCM shouldn't co-exist so we may need to move hasAnyGcmSpecificDeclaration() out too as part of the interface?  (canHandlePush?)

Reply:
That sounds good! I will do it. In the future we might even make the handler public so people can provide their adapters like parse server.

By the way, is parse-server ready to send FCM-like messages? AFAIK currently the handlers here need to update the ParseInstallation `pushType` to tell parse server adapter how to send the push. What happens when we start setting `pushType = 'fcm'` ?

Reply:
Yeah parse server may need to be upgraded.  But the push tokens should be interoperable so hopefully t

Reply:
FCM is already handled:

https://github.com/parse-server-modules/parse-server-push-adapter/blob/dfa75b51eb3fb909572ad7ecaa11f29a9ecc917f/src/ParsePushAdapter.js#L15

Reply:
I do think that the deviceType (https://github.com/parse-community/Parse-SDK-Android/blob/6e8804cd721513dff78c68ea9b99f27830e41e3c/Parse/src/main/java/com/parse/ParseInstallation.java#L288) is what gets used over GCM/FCM right now..

See https://github.com/parse-server-modules/parse-server-push-adapter/blob/ef59a8c8eec303ad9547e02223765a3699cc5032/src/PushAdapterUtils.js#L20

Reply:
The actual send implementation is still GCM based though? https://github.com/parse-server-modules/parse-server-push-adapter/blob/master/src/GCM.js I don't know if that's a problem and how we should manage